### PR TITLE
Add S3 Download Demo From C-SDK

### PR DESCRIPTION
**Pull request by sukhmanm**
Adding demo to demonstrate the use of pre-signed URLs to connect and download from a S3 bucket. Sends multiple GET requests synchronously to download a file in chunks using range headers.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: demos/coreHTTP/http_demo_s3_download.c [hunk 1]
```diff
@@ -0,0 +1,729 @@
...
+
+/* Common HTTP demo utilities. */
+#include "http_demo_utils.h"
+
+/* HTTP API header. */
+#include "core_http_client.h"
+
+/*------------- Demo configurations -------------------------*/
+
+/* Check that TLS port of the server is defined. */
```
Comment (sarenameas) [inline demos/coreHTTP/http_demo_s3_download.c:72]:
```suggestion
/* Check that the TLS port of the server is defined. */
```

File: demos/coreHTTP/http_demo_s3_download.c [hunk 2]
```diff
@@ -0,0 +1,729 @@
...
+#ifndef democonfigROOT_CA_PEM
+    #error "Please define a democonfigROOT_CA_PEM."
+#endif
+
+/* Check that the pre-signed GET URL is defined. */
+#ifndef democonfigS3_PRESIGNED_GET_URL
+    #define democonfigS3_PRESIGNED_GET_URL    "GET-URL"
+#endif
+
+/* Check that transport timeout for transport send and receive is defined. */
```
Comment (sarenameas) [inline demos/coreHTTP/http_demo_s3_download.c:87]:
```suggestion
/* Check that the transport timeout for transport send and receive functions is defined. */
```

File: demos/coreHTTP/http_demo_s3_download.c [hunk 3]
```diff
@@ -0,0 +1,729 @@
...
+            LogError( ( "Failed to connect to HTTP server %s.",
+                        pxServerHost ) );
+        }
+    }
+
+    /* Define the transport interface. */
+    if( xDemoStatus == pdPASS )
+    {
+        /* Set a flag indicating a TLS connection exists. This is done to
+         * disconnect if the loop exits before disconnection happens. */
```
Comment (sarenameas) [inline demos/coreHTTP/http_demo_s3_download.c:676]: We can take out this loop comment from the demos.

Comment (sukhmanm) [reply demos/coreHTTP/http_demo_s3_download.c]: Thanks for catching that.

**Review by sarenameas — DISMISSED** [review]
Contingent on passing on the 5 reference platforms in CI.

**Review by sarenameas — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**